internal/provider/auth.go

```diff
@@ -278,34 +278,39 @@ func (l *AuthLoginCommon) init(d *schema.ResourceData) (string, map[string]inter
 
 type authDefault struct {
 	field      string
-	envVar     string
+	envVars    []string
 	defaultVal string
 }
 
 type authDefaults []authDefault
 
-func (l *AuthLoginCommon) setDefaultFields(d *schema.ResourceData, defaults authDefaults) error {
+func (l *AuthLoginCommon) setDefaultFields(d *schema.ResourceData, defaults authDefaults, params map[string]interface{}) error {
 	for _, f := range defaults {
 		if _, ok := l.getOk(d, f.field); !ok {
-
 			// if field is unset in the config, check env
-			defaultValue := os.Getenv(f.envVar)
-			if defaultValue == "" {
-				defaultValue = f.defaultVal
+			for _, envVar := range f.envVars {
```

Comment:
This allows us to pass multiple options for the default environment variable and take the first value we find. Similar to what `MultiEnvDefaultFunc` is achieving [here](https://github.com/hashicorp/terraform-provider-vault/pull/2097/files#diff-a8ce64c79c976a0549f2566d9c67999b437f3703011fb0b2fc81bbcd766bd7fbL105)